packages/libs/src/sdk/api/ResolveApi.ts

```diff
@@ -41,9 +39,14 @@ export class ResolveApi extends BaseAPI {
       query: queryParameters
     })
     return new JSONApiResponse(response, (json) => {
-      if (instanceOfTrackResponse(json)) {
+      const data = json?.data ?? {}
+      if ('title' in data) {
```

Comment:
why are these changes necessary? i would prefer typeguards to ad-hoc `x in y` stand-ins for type inferences.

Reply:
like even if you pulled out
```
        Array.isArray(data) &&
        data.length > 0 &&
        'playlist_name' in data[0]
```
to a shared typeguard function in common or something that would be better. feels bad to make everyone have to remember to do this check when trying to determine if an object is a playlist.

Reply:
i guess the array makes it hard but you get my drift.

Reply:
instanceOfTrackResponse returns `true` always and is generated, so idk what is up there, but ill make this a helper so it's more clean. i think it's a flaskrestx shortcoming, but just dont want to waste time there. resolve is currently just straight up broken - always returns a track shape even if a playlist or a user